src/radical/pilot/agent/launch_method/base.py

```diff
@@ -53,56 +50,61 @@
 #
 class LaunchMethod(object):
 
-    # List of environment variables that designated Launch Methods should export
-    # FIXME: we should find out what env vars are changed or added by
-    #        td.pre_exec, and then should also export those.  That would make
-    #        our launch script ore complicated though...
-    EXPORT_ENV_VARIABLES = [
-        'LD_LIBRARY_PATH',
-        'PATH',
-        'PYTHONPATH',
-        'OMP_NUM_THREADS',
-        'RP_AGENT_ID',
-        'RP_GTOD',
-        'RP_PILOT_ID',
-        'RP_PILOT_STAGING',
-        'RP_PROF',
-        'RP_SESSION_ID',
-        'RP_SPAWNER_ID',
-        'RP_TMP',
-        'RP_TASK_ID',
-        'RP_TASK_NAME',
-        'RP_PILOT_SANDBOX',
-        'RADICAL_BASE'
-    ]
-
     MPI_FLAVOR_OMPI    = 'OMPI'
     MPI_FLAVOR_HYDRA   = 'HYDRA'
     MPI_FLAVOR_UNKNOWN = 'unknown'
 
+
     # --------------------------------------------------------------------------
     #
-    def __init__(self, name, cfg, session):
-
-        self.name     = name
-        self._cfg     = cfg
-        self._session = session
-        self._log     = self._session._log    # pylint: disable=protected-access
-        self._log.debug('create LaunchMethod: %s', type(self))
-
-        # A per-launch_method list of env vars to remove from the Task env
-        self.env_removables = []
-
-        self._configure()
+    def __init__(self, name, lm_cfg, cfg, log, prof):
+
+        log.debug('===== lm base init start')
+        self.name    = name
+        self._lm_cfg = cfg
```

Comment:
seems that this attribute `self._lm_cfg` is never used (and should input parameter `lm_cfg` be assigned?)

Reply:
Thanks, I fixed the assignment.  Not sure if we need to keep it around, it's indeed not used, but we usually keep configs in other classes also, so... *shrug*